### PR TITLE
Feature/sensor expansion

### PR DESCRIPTION
**Pull request by naveenoid**
Generic Sensor from URDF. 
First commits will focus on new sensor addition and on structure of URDF parsing retaining backward compatibility to gazebo extension FTSensors. Subsequent will focus on implementing  http://wiki.ros.org/urdf/XML/sensor/proposals

### CODE REVIEW

**Comment by naveenoid** [comment]
Travis fails with this error : The command "sudo apt-get install cmake" failed and exited with 100 during . 

I have no idea why. I will proceed with subsequent commits anyhow.

**Comment by naveenoid** [comment]
@jeljaik : just to let you know sensor URDF processing will be done soon. This PR will consolidate the commits related to that.

**Comment by traversaro** [comment]
Travis problems seem related to networks errore, please restart the build in those cases.

**Comment by naveenoid** [comment]
<parent> modified to include an attribute of either link or joint. <pose> tag changed to <origin>

**Comment by traversaro** [comment]
I have added the option to run the tests (for now just the model_io one) with Valgrind in https://github.com/robotology/idyntree/commit/1c489bea345532daeaaeab25537dc19a46d59c61 . You can enable this tests by marking to ON the IDYNTREE_RUN_VALGRIND_TESTS cmake option in the iDynTree build. 

To run only the valgrind tests, run:
~~~
ctest -VV -R memcheck
~~~

Running this in this branch results in this error:
~~~
14: Test command: /usr/bin/valgrind "--leak-check=full" "--error-exitcode=1" "./URDFGenericSensorImportUnitTest"
14: Test timeout computed to be: 1500
14: ==25215== Memcheck, a memory error detector
14: ==25215== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
14: ==25215== Using Valgrind-3.10.0.SVN and LibVEX; rerun with -h for copyright info
14: ==25215== Command: ./URDFGenericSensorImportUnitTest
14: ==25215== 
14: Generic Sensor test running (one Link):
14: Sensor list created from URDF. num accel : 2, num gyro : 1
14: Generic Sensor test running (two Link):
14: Sensor list created from URDF. num accel : 2, num gyro : 1
14: Generic Sensor test just ran
14: ==25215== 
14: ==25215== HEAP SUMMARY:
14: ==25215==     in use at exit: 1,785 bytes in 24 blocks
14: ==25215==   total heap usage: 1,770 allocs, 1,746 frees, 132,687 bytes allocated
14: ==25215== 
14: ==25215== 297 (16 direct, 281 indirect) bytes in 1 blocks are definitely lost in loss record 7 of 12
14: ==25215==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
14: ==25215==    by 0x407E94C: iDynTree::genericSensorsListFromURDFString(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:252)
14: ==25215==    by 0x407E75F: iDynTree::genericSensorsListFromURDF(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:215)
14: ==25215==    by 0x4082017: iDynTree::genericSensorsListFromURDF(std::string const&, iDynTree::SensorsList&) (URDFGenericSensorsImport.cpp:33)
14: ==25215==    by 0x4013C0: checkURDF(std::string, unsigned int, unsigned int) (URDFGenericSensorImportUnitTest.cpp:25)
14: ==25215==    by 0x4016C9: main (URDFGenericSensorImportUnitTest.cpp:38)
14: ==25215== 
14: ==25215== 297 (16 direct, 281 indirect) bytes in 1 blocks are definitely lost in loss record 8 of 12
14: ==25215==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
14: ==25215==    by 0x407E94C: iDynTree::genericSensorsListFromURDFString(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:252)
14: ==25215==    by 0x407E75F: iDynTree::genericSensorsListFromURDF(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:215)
14: ==25215==    by 0x4082017: iDynTree::genericSensorsListFromURDF(std::string const&, iDynTree::SensorsList&) (URDFGenericSensorsImport.cpp:33)
14: ==25215==    by 0x4013C0: checkURDF(std::string, unsigned int, unsigned int) (URDFGenericSensorImportUnitTest.cpp:25)
14: ==25215==    by 0x40174F: main (URDFGenericSensorImportUnitTest.cpp:41)
14: ==25215== 
14: ==25215== 595 (32 direct, 563 indirect) bytes in 2 blocks are definitely lost in loss record 11 of 12
14: ==25215==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
14: ==25215==    by 0x407E901: iDynTree::genericSensorsListFromURDFString(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:245)
14: ==25215==    by 0x407E75F: iDynTree::genericSensorsListFromURDF(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:215)
14: ==25215==    by 0x4082017: iDynTree::genericSensorsListFromURDF(std::string const&, iDynTree::SensorsList&) (URDFGenericSensorsImport.cpp:33)
14: ==25215==    by 0x4013C0: checkURDF(std::string, unsigned int, unsigned int) (URDFGenericSensorImportUnitTest.cpp:25)
14: ==25215==    by 0x4016C9: main (URDFGenericSensorImportUnitTest.cpp:38)
14: ==25215== 
14: ==25215== 596 (32 direct, 564 indirect) bytes in 2 blocks are definitely lost in loss record 12 of 12
14: ==25215==    at 0x4C2B0E0: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
14: ==25215==    by 0x407E901: iDynTree::genericSensorsListFromURDFString(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:245)
14: ==25215==    by 0x407E75F: iDynTree::genericSensorsListFromURDF(KDL::CoDyCo::UndirectedTree&, std::string) (urdf_generic_sensor_import.cpp:215)
14: ==25215==    by 0x4082017: iDynTree::genericSensorsListFromURDF(std::string const&, iDynTree::SensorsList&) (URDFGenericSensorsImport.cpp:33)
14: ==25215==    by 0x4013C0: checkURDF(std::string, unsigned int, unsigned int) (URDFGenericSensorImportUnitTest.cpp:25)
14: ==25215==    by 0x40174F: main (URDFGenericSensorImportUnitTest.cpp:41)
14: ==25215== 
14: ==25215== LEAK SUMMARY:
14: ==25215==    definitely lost: 96 bytes in 6 blocks
14: ==25215==    indirectly lost: 1,689 bytes in 18 blocks
14: ==25215==      possibly lost: 0 bytes in 0 blocks
14: ==25215==    still reachable: 0 bytes in 0 blocks
14: ==25215==         suppressed: 0 bytes in 0 blocks
14: ==25215== 
14: ==25215== For counts of detected and suppressed errors, rerun with: -v
14: ==25215== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
2/2 Test #14: memcheck_UnitTestURDFGenericSensorImport ...***Failed    2.73 sec
~~~

Directly related to https://github.com/robotology/idyntree/pull/89#commitcomment-14390229 .

File: model_io/urdf/src/urdf_generic_sensor_import.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+            double x  = atof(xyzElems[0].c_str());
+            double y = atof(xyzElems[1].c_str());
+            double z   = atof(xyzElems[2].c_str());
+            iDynTree::Position pos(x,y,z);
+            newGenericSensor.sensorPose = iDynTree::Transform(rot,pos); 
+        }
+        else
+        {
+            // default value
+            newGenericSensor.sensorPose = iDynTree::Transform();
```
Comment (traversaro) [inline model_io/urdf/src/urdf_generic_sensor_import.cpp]: There is also a function `iDynTree::Transform::Identity()` to express more clearly the the default transform is the identity.

**Comment by traversaro** [comment]
All tests passing (the travis failure is related to master failure introduced in https://github.com/robotology/idyntree/commit/bb3b62b5bd27310c956b18916300549419091f4e and fixed with https://github.com/robotology/idyntree/commit/e7f1b197d8b7631b03282487da085908974c6073), merging. :+1:

**Pull request merged**